packages/peregrine/lib/talons/ProductFullDetail/useProductFullDetail.js

```diff
@@ -203,10 +203,20 @@ export const useProductFullDetail = props => {
             appendOptionsToPayload(payload, optionSelections, optionCodes);
         }
 
+        let addItemMutation;
+
+        // Prepare the params to add and use the proper mutation.
+        if (productType === 'SimpleProduct') {
+            addItemMutation = addSimpleProductToCart;
+        } else if (productType === 'ConfigurableProduct') {
+            addItemMutation = addConfigurableProductToCart;
+        } else {
+            throw new Error('Unsupported product type. Cannot add to cart.');
```

Comment:
This is also now a unhandled exception. Since we know the add will fail, why don't we just disable the button, and `console.error`. If you have a more user friendly way to handle this, open to suggestions.

Reply:
I'm not expecting this case to occur -- we only use `Simple` and `Configurable` in our storefront.

That said, I'll disable the button for now so it's impossible to attempt. I'll log something with it too.